doge_datagen/kafka_sink_factory.py

```diff
@@ -61,20 +61,30 @@ def close(self):
 
 class KafkaSinkFactory(object):
     def __init__(self,
-                 bootstrap_servers: Iterable[str],
-                 client_id: str,
-                 buffer_size=100000):
+                bootstrap_servers:Iterable[str],
```

Comment:
This could be:
```
def __init__(self,
                bootstrap_servers:Iterable[str] = None,
                                client_id:str = None,
                conf:dict=None,
                buffer_size=100000
                ):
```
So it will still be backward compatible but you can just init it with 
```
KafkaSinkFactory(conf={..})
```

Reply:
init changed

Reply:
I meant to make `bootstrap_servers` and `client_id` optional parameters. So when using conf we can just pass 
```
KafkaSinkFactory(conf={..})
```

Now we have to pass `boostrap_servers` and `client_id` even thou we want to override it using `conf`

Reply:
If we set variables in __init__ I would need to change the sequence to :
def __init__(self,
                conf:dict=None,
                bootstrap_servers:Iterable[str] = None,
                client_id:str = None,
                buffer_size=100000
                ):

otherwise user should always pass bootstrap and client_id when init the object
So my suggestion was to keep bootstrap_servers and client_id (which are obligatory for Producer) and keep additional configuration in conf dict. 
What do you think?

Reply:
I don't get it.
You can do:
```
def __init__(self,
                 bootstrap_servers: Iterable[str] = None,
                 client_id: str = None,
                 buffer_size=100000,
                 conf=None):
```
Order of previous list of parameters is preserve so its compatible and new parameter is optional. BTW. don't pass mutable values as default value as in Python it is evaluated once and if you would add something to it it would be passed as default to all next instances.

And you can execute as before: 
```
KafkaSinkFactory(["localhost:9092"], "client1")
```
And with conf dict using named parameter:
```
KafkaSinkFactory(conf={"bootstrap": "localhost:9092", "client.id": "client1"})
```

Reply:
Yes, correct. Changed accordingly